src/gobprepare/importers/api_importer.py

```diff
@@ -0,0 +1,184 @@
+        self.host = get_host()
+        self._dst_connection = dst_connection
+        self.query = query
+        self.schema = config['schema']
+        self.table_name = config['destination']
+        self.description = config['description']
+        self.meta_type = config['meta_type']
+
+    def import_api(self):
+        """Make API calls and import data into the database.
+
+        Make call to GraphQL Streaming API to fetch data using query.
+        Make call to GraphQL API to fetch meta data. Meta data will be using to create
+        database table.
+
+        Import data into the Postgres database.
+
+        :return: Number of records imported.
+        """
+        entities = self.get_entities()
+        meta_data = self.get_meta_data()
+        return self.import_data(entities, meta_data)
+
+    def get_entities(self):
+        """Get entities from GraphQL API.
+
+        :return: Iterator to entities.
+        """
+        return GraphQL(self.host, STREAMING_GRAPHQL_ENDPOINT, self.query)
+
+    def get_meta_data(self):
+        """Get meta data from GraphQL API.
+
+        :return: Iterator to meta data.
+        """
+        meta_query = self.get_meta_query(self.meta_type)
+        return GraphQL(self.host, GRAPHQL_ENDPOINT, meta_query)
+
+    def import_data(self, entities, meta_data):
+        """Re-create table and import data into the Postgres database.
+
+        :param entities: Entities to import.
+        :param meta_data: Meta data to create a table.
+        :return: Number of records imported.
+        """
+        create_schema(self._dst_connection, self.schema)
+
+        # clone entities to get field name from the first record
+        entities_clone, entities = itertools.tee(entities, 2)
+        field_names = self.get_field_names(entities_clone)
+        meta_fields = self.get_meta_fields(meta_data)
+        self.create_table(meta_fields, field_names)
+        return self.import_entities(entities, field_names)
+
+    @staticmethod
+    def get_meta_query(meta_type):
+        """Construct meta data GraphQL query for a certain entity (meta_type).
+
+        :param meta_type: GraphQL meta type name.
+        :return: Meta data GraphQL query.
+        """
+        return '''
+{
+  __type(name: "%s") {
+    name
+    fields {
+      name
+      description
+      type {
+        name
+      }
+    }
+  }
+}
+''' % meta_type
+
+    def get_meta_fields(self, meta_data):
+        """Get meta fields from meta data.
+
+        :param meta_data: Meta data.
+        :return: Dict of meta fields.
+        """
+        meta_fields = {}
+        for meta in meta_data:
+            meta_fields.update({field["name"]: field for field in meta["data"]["__type"]["fields"]})
+        return meta_fields
+
+    def get_field_names(self, entities):
+        """Get field names from a first data record.
+
+        :param entities: Data entities.
+        :return: List of field names.
+        """
+        first_record = next(entities)
```

Comment:
What happens if no data can be read (GOB is still empty). Having GOB data is a prerequisite to this import so it should either be asserted with a logical error description or otherwise be handled. It should also be documented in the README

Reply:
Can you add this as a requirement to the README. Prepare can only run when this precondition is met